### PR TITLE
[ACA-2619][ACA-2616] a11y fixes for Share Link dialog

### PR DESCRIPTION
**Pull request by Steady5063**
**Please check if the PR fulfills these requirements**

> - [ X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [X ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
After switching on the expiration toggle, the calendar appears. This is not proper from an accessibility standpoint. 

**What is the new behaviour?**
The field will be disabled if there is not date added into it. and when the switch toggle is turned on the field will become enabled. The action of the calendar appearing on click has been removed. 

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/content-services/src/lib/content-node-share/content-node-share.dialog.ts [hunk 1]
```diff
@@ -126,6 +126,12 @@ export class ShareDialogComponent implements OnInit, OnDestroy {
         }
     }
 
+    ngAfterViewInit() {
```
Comment (eromano) [inline lib/content-services/src/lib/content-node-share/content-node-share.dialog.ts]: all this change afterViewInit should be avoided

Comment (DenysVuika) [reply lib/content-services/src/lib/content-node-share/content-node-share.dialog.ts]: I have removed that part

File: e2e/content-services/share-file/share-file.e2e.ts [hunk 2]
```diff
@@ -142,18 +144,11 @@ describe('Share file', () => {
             await BrowserActions.closeMenuAndDialogs();
         });
 
-        it('[C286578] Should disable today option in expiration day calendar', async () => {
```
Comment (adinapitul) [inline e2e/content-services/share-file/share-file.e2e.ts]: why are you removing this test?

Comment (DenysVuika) [reply e2e/content-services/share-file/share-file.e2e.ts]: it's a copy-pasted one, the same test is above :)

Comment (adinapitul) [reply e2e/content-services/share-file/share-file.e2e.ts]: hehe 😄

**Pull request merged**